sdk/core/core-rest-pipeline/src/policies/formDataPolicy.ts

```diff
@@ -45,40 +51,52 @@ function wwwFormUrlEncode(formData: FormDataMap): string {
   return urlSearchParams.toString();
 }
 
-async function prepareFormData(formData: FormDataMap, request: PipelineRequest): Promise<void> {
-  const requestForm = new FormData();
-  for (const formKey of Object.keys(formData)) {
-    const formValue = formData[formKey];
-    if (Array.isArray(formValue)) {
-      for (const subValue of formValue) {
-        requestForm.append(formKey, subValue);
-      }
-    } else {
-      requestForm.append(formKey, formValue);
-    }
-  }
+const encoder = new TextEncoder();
 
-  request.body = requestForm;
+async function prepareFormData(formData: FormDataMap, request: PipelineRequest): Promise<void> {
+  // clear request.formData
   request.formData = undefined;
```

Comment:
Should we warn or throw if it is already set?

Reply:
are you referring to the body or to `formData` here? I agree we should make it throw if a body is already there. I updated it to do that just now.

Reply:
I meant `formData` but good call on `body`.

Reply:
`request.formData` is always going to be set here -- and that's expected -- since we check that it's not undefined further up in the policy itself. We can move actually clearing it to the policy itself, I think that would be a bit clearer (and that's what we do in the case of urlencoded forms).